File: tests/output/csharp/src/Clients/TopModel.Sample.Clients.Db/Models/NombreVuesBase.cs
////
//// ATTENTION CE FICHIER EST GENERE AUTOMATIQUEMENT !
////

using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Kinetix.Modeling.Annotations;
using TopModel.Sample.Restaurant.Models;

namespace TopModel.Sample.Clients.Db.Models;

public record NombreVuesBase
{
    /// <summary>
    /// Nombre de vues de l'avis (calculé).
    /// </summary>
    [NotMapped]
    [Required]
    [Domain(Domains.Quantite)]
    public int? NombreVues { get; init; } = 0;
}

File: tests/output/jpa/src/main/java/topmodel/test/NombreVuesBase.java
package topmodel.test;

import jakarta.persistence.Transient;

public class NombreVuesBase {

    /**
     * Nombre de vues de l'avis (calculé).
     */
    @Transient
    private Integer nombreVues = 0;

    /**
     * Getter for nombreVues.
     *
     * @return value of {@link #nombreVues nombreVues}.
     */
    public Integer getNombreVues() {
        return this.nombreVues;
    }

    /**
     * Set the value of {@link #nombreVues nombreVues}.
     * 
     * @param nombreVues value to set.
     */
    public void setNombreVues(Integer nombreVues) {
        this.nombreVues = nombreVues;
    }
}
